Modules/Sources/WordPressKit/ReaderFeed.swift

```diff
@@ -30,61 +66,52 @@ public struct ReaderFeed: Decodable {
     }
 
     public init(from decoder: Decoder) throws {
-        // We have to manually decode the feed from the JSON, for a couple of reasons:
-        // - Some feeds have no `icon` dictionary
-        // - Some feeds have no `data` dictionary
-        // - We want to decode whatever we can get, and not fail if neither of those exist
-        let rootContainer = try decoder.container(keyedBy: CodingKeys.self)
-
-        var feedURL = try? rootContainer.decodeIfPresent(URL.self, forKey: .url)
-        var title = try? rootContainer.decodeIfPresent(String.self, forKey: .title)
-        feedID = try? rootContainer.decode(String.self, forKey: .feedID)
-        blogID = try? rootContainer.decode(String.self, forKey: .blogID)
-
-        var feedDescription: String?
-        var blavatarURL: URL?
-
-        // Try to parse both site and feed data from meta.data
-        do {
-            let metaContainer = try rootContainer.nestedContainer(keyedBy: MetaKeys.self, forKey: .meta)
-            let dataContainer = try metaContainer.nestedContainer(keyedBy: DataKeys.self, forKey: .data)
-
-            let siteData = try? dataContainer.decode(SiteOrFeedData.self, forKey: .site)
-            let feedData = try? dataContainer.decode(SiteOrFeedData.self, forKey: .feed)
-
-            // Use data from either source, preferring site data when both are available
-            feedDescription = siteData?.description ?? feedData?.description
-            blavatarURL = siteData?.iconURL ?? feedData?.iconURL
-
-            // Fixes CMM-1002: in some cases, the backend fails to embed certain fields
-            // directly in the feed object
-            if feedURL == nil {
-                feedURL = siteData?.url ?? feedData?.url
-            }
-            if title == nil {
-                title = siteData?.title ?? feedData?.title
-            }
-        } catch {
+        let root = try decoder.container(keyedBy: CodingKeys.self)
+        if let meta = try? root.nestedContainer(keyedBy: MetaKeys.self, forKey: .meta),
+           let data = try? meta.nestedContainer(keyedBy: DataKeys.self, forKey: .data) {
+            self.feed = try? data.decode(FeedData.self, forKey: .feed)
+            self.site = try? data.decode(SiteData.self, forKey: .site)
         }
+    }
+}
+
+// MARK: - Feed Data
 
-        self.url = feedURL
-        self.title = title
-        self.feedDescription = feedDescription
-        self.blavatarURL = blavatarURL
+/// Represents feed-specific data from meta.data.feed
+private struct FeedData: Decodable {
+    let feedID: String?
+    let blogID: String?
+    let name: String?
+    let url: URL?
```

Comment:
During testing the JSON parsing above, I found out by accident that this `url` can be an empty string, which causes JSONDecoder to throw an error, which is silenced at the moment because of the `try?` parsing code above.

You can search "WordPress.com News" and check out the "wpsites.net" site in the response.

Reply:
I think we can change the `URL` properties to be `String`, and turn the string to `URL` in the ReaderFeed public API, which are all optional anyway.

Reply:
I added a custom `init(from decoder:)` for `FeedData` (same as `SiteData` already has). I don't trust every field to be the correct format, so it all `try?`s now. It will catch the `URL` issue too.